src/platform/backends/qemu/dnsmasq_server.cpp

```diff
@@ -105,3 +135,21 @@ void mp::DNSMasqServer::release_mac(const std::string& hw_addr)
 
     dhcp_release.waitForFinished();
 }
+
+void mp::DNSMasqServer::check_dnsmasq_running()
+{
+    auto dnsmasq_pid = get_dnsmasq_pid(pid_file_path);
+    if (kill(dnsmasq_pid, 0) == 0)
+        return;
+
+    // exit_code == 2 signifies dnsmasq network-related error. See `man dnsmasq`.
+    throw std::runtime_error(
+        fmt::format("Multipass dnsmasq is not running.{}",
+                    (dnsmasq_exit_state.exit_code == 2) ? " Ensure nothing is using port 53." : ""));
+}
+
+void mp::DNSMasqServer::start_dnsmasq()
+{
+    dnsmasq_cmd = make_dnsmasq_process(data_dir, bridge_name, pid_file_path, subnet);
+    dnsmasq_exit_state = dnsmasq_cmd->execute();
```

Comment:
Nitpick: `dnsmasq_exit_state` contains the return code of the now daemonized dnsmasq process - could you rename to "dnsmasq_daemon_fork_state" or something like that?

You're not handling cases if dnsmasq binary isn't found. Wouldn't hurt...

Reply:
Sure, I can fix up both of those :grin:

Reply:
Fixed both of these nits.